src/Report/EnabledMembers.php

```diff
@@ -0,0 +1,206 @@
+
+namespace SilverStripe\MFA\Report;
+
+use SilverStripe\Forms\CheckboxField;
+use SilverStripe\Forms\DropdownField;
+use SilverStripe\Forms\FieldList;
+use SilverStripe\Forms\NumericField;
+use SilverStripe\Forms\TextField;
+use SilverStripe\MFA\Extension\MemberExtension;
+use SilverStripe\MFA\Service\MethodRegistry;
+use SilverStripe\ORM\ArrayList;
+use SilverStripe\ORM\Filterable;
+use SilverStripe\ORM\SS_List;
+use SilverStripe\Reports\Report;
+use SilverStripe\Security\Member;
+use SilverStripe\View\ArrayData;
+
+if (!class_exists(Report::class)) {
+    return;
+}
+
+class EnabledMembers extends Report
+{
+    protected $title = 'Multi factor authentication status of members';
+
+    protected $description = 'Identifies all members on this site, with columns added to indicate the status each has'
+        . ' in regards to multi factor authentication method registration. E.g. how many methods are set up by that'
+        . ' member, which method types, and who has opted not to set up MFA (if this option is enabled for the site)';
+
+    protected $dataClass = Member::class;
+
+    public function title()
+    {
+        return _t(__CLASS__ . '.TITLE', parent::title());
+    }
+
+    public function description()
+    {
+        return _t(__CLASS__ . '.DESCRIPTION', parent::description());
+    }
+
+    /**
+     * @param array $params
+     */
+    public function sourceRecords($params)
+    {
+        $methods = $this->getMethodClassToTitleMapping();
+
+        // Any GridFieldAction will submit all form inputs, regardless of whether or not they have valid values.
+        // This results in zero results when no filters have been set, as the query then filters for empty string
+        // on all parameters. `stlen` is a safe function, as all user input comes through to PHP as strings.
+        $params = array_filter($params, 'strlen');
+
+        // Having a WHERE limits the results, which means the the COUNT will be off (GROUP BY applied after WHERE)
+        // So clip the count filter and apply it after the database query, only if Methods filter is also set
+        // this way we can delegate to the database where possible, which should be faster.
+        if (isset($params['Count'])) {
+            $desiredCount = (int)$params['Count'];
+            if (isset($params['Methods'])) {
+                $countFilter = $desiredCount;
+                unset($params['Count']);
+            } elseif ($desiredCount) {
+                // When querying SQL COUNT() - this will always factor in the backup method, which is excluded in the
+                // secondary query below to get the list of method names the user has registered. So to compensate we
+                // need to bump up the entered number by 1.
+                $params['Count'] = $desiredCount + 1;
+            }
+        }
+
+        $filteredList = ArrayList::create([]);
+        $backupClass = get_class(singleton(MethodRegistry::class)->getBackupMethod());
```

Comment:
I don't like this very much... but I realise we don't have a great API to return it, other than using the config layer, which isn't the only logic determining this...

Reply:
Can you expand on this please?

As far as I understand it, even though Injector will create the backup method (meaning it could be a different class to the service specified), the concrete class name will be what is stored in the database.
Only if these two become out of sync will it be an issue, but this is true for all occurrences, as you've alluded to in your comment.

Indeed I think this is the best that can be done given the tools available at the moment.

Reply:
>Can you expand on this please?

We _could_ add a method for it, so you could do `MethodRegistry::singleton()->getBackupMethodClass()`, but I'm not suggesting we need to do it, more just pointing out that this line of code isn't ideal. It's also mentioned [further down](https://github.com/silverstripe/silverstripe-mfa/pull/299#discussion_r310407498) that centralising it in a report method instead might be a middleground, which I think you might've done, so we can resolve now!